Move MongoDB image in `debezium-testing-testcontainers` to quay registry

## Context

In some cases, if we use the docker image defined in `debezium-testing-testcontainers` in the [ImageNames.java](https://github.com/debezium/debezium/blob/8e6f0810b55b0b041086e4cbbeb7a84174628ccb/debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ImageNames.java#L22), it's impossible to pull the image due to rate limitation towards `mirror.gcr.io` with the following error:

```txt
[ERROR] io.quarkus.debezium.mongodb.deployment.MongoDbDeploymentExtensionTest -- Time elapsed: 1.969 s <<< ERROR!
java.lang.IllegalStateException: java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mirror.gcr.io/library/mongo:6.0
        at io.debezium.testing.testcontainers.util.MoreStartables.deepStartSync(MoreStartables.java:35)
        at io.debezium.testing.testcontainers.MongoDbReplicaSet.start(MongoDbReplicaSet.java:286)
        at io.quarkus.debezium.mongodb.deployment.MongoDbTestResource.start(MongoDbTestResource.java:41)
        at io.quarkus.debezium.mongodb.deployment.MongoDbDeploymentExtensionTest.init(MongoDbDeploymentExtensionTest.java:25)
Caused by: java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mirror.gcr.io/library/mongo:6.0
        at java.base/java.util.concurrent.CompletableFuture.wrapInExecutionException(CompletableFuture.java:347)
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:442)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2119)
        at io.debezium.testing.testcontainers.util.MoreStartables.deepStartSync(MoreStartables.java:32)
        ... 3 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mirror.gcr.io/library/mongo:6.0
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:346)
        at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:317)
        at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:833)
        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:528)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
        at java.base/java.lang.Thread.run(Thread.java:1447)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
        at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:331)
        ... 6 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
        at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:551)
        at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:341)
        at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
        ... 7 more
Caused by: com.github.dockerjava.api.exception.BadRequestException: Status 400: {"message":"all predefined address pools have been fully subnetted"}
```

## Changes

We should change the official docker image for test in images that are in the `quay` registry. Furthermore as we already have a `mongodb` image which is CDC-ready, we should check if it's necessary a vanilla image or the tweaked one for the tests

--- a/.github/workflows/connector-mongodb-workflow.yml
+++ b/.github/workflows/connector-mongodb-workflow.yml
@@ -17,7 +17,7 @@ jobs:
       max-parallel: ${{ inputs.max-parallel }}
       fail-fast: false
       matrix:
-        server-version: [ "5.0", "8.0" ]
+        server-version: [ "7.0", "8.0" ]
     name: MongoDB - ${{ matrix.server-version }}
     runs-on: ubuntu-latest
 

--- a/debezium-connector-jdbc/src/test/resources/testcontainers.properties
+++ b/debezium-connector-jdbc/src/test/resources/testcontainers.properties
@@ -1,1 +1,2 @@
+hub.image.name.prefix=quay.io/
 ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-jdbc/src/test/resources/testcontainers.properties
+++ b/debezium-connector-jdbc/src/test/resources/testcontainers.properties
@@ -1,1 +1,1 @@
-hub.image.name.prefix=mirror.gcr.io/
+ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-mariadb/pom.xml
+++ b/debezium-connector-mariadb/pom.xml
@@ -186,7 +186,7 @@
         <!--
         Specify the properties for the various Docker containers.
         -->
-        <mariadb.server.image.source>mirror.gcr.io/library/mariadb</mariadb.server.image.source>
+        <mariadb.server.image.source>quay.io/debezium/official-mariadb</mariadb.server.image.source>
         <version.mariadb.server>11.4.3</version.mariadb.server>
         <mariadb.user>mysqluser</mariadb.user>
         <mariadb.password>mysqlpw</mariadb.password>

--- a/debezium-connector-mariadb/src/test/resources/testcontainers.properties
+++ b/debezium-connector-mariadb/src/test/resources/testcontainers.properties
@@ -1,1 +1,2 @@
+hub.image.name.prefix=quay.io/
 ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-mariadb/src/test/resources/testcontainers.properties
+++ b/debezium-connector-mariadb/src/test/resources/testcontainers.properties
@@ -1,1 +1,1 @@
-hub.image.name.prefix=mirror.gcr.io/
+ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-mongodb/src/test/resources/testcontainers.properties
+++ b/debezium-connector-mongodb/src/test/resources/testcontainers.properties
@@ -1,1 +1,2 @@
+hub.image.name.prefix=quay.io/
 ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-mongodb/src/test/resources/testcontainers.properties
+++ b/debezium-connector-mongodb/src/test/resources/testcontainers.properties
@@ -1,1 +1,1 @@
-hub.image.name.prefix=mirror.gcr.io/
+ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-mysql/src/test/resources/testcontainers.properties
+++ b/debezium-connector-mysql/src/test/resources/testcontainers.properties
@@ -1,1 +1,2 @@
+hub.image.name.prefix=quay.io/
 ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-mysql/src/test/resources/testcontainers.properties
+++ b/debezium-connector-mysql/src/test/resources/testcontainers.properties
@@ -1,1 +1,1 @@
-hub.image.name.prefix=mirror.gcr.io/
+ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-oracle/src/test/resources/testcontainers.properties
+++ b/debezium-connector-oracle/src/test/resources/testcontainers.properties
@@ -1,1 +1,2 @@
+hub.image.name.prefix=quay.io/
 ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-oracle/src/test/resources/testcontainers.properties
+++ b/debezium-connector-oracle/src/test/resources/testcontainers.properties
@@ -1,1 +1,1 @@
-hub.image.name.prefix=mirror.gcr.io/
+ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-postgres/src/test/resources/testcontainers.properties
+++ b/debezium-connector-postgres/src/test/resources/testcontainers.properties
@@ -1,1 +1,2 @@
+hub.image.name.prefix=quay.io/
 ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-connector-postgres/src/test/resources/testcontainers.properties
+++ b/debezium-connector-postgres/src/test/resources/testcontainers.properties
@@ -1,1 +1,1 @@
-hub.image.name.prefix=mirror.gcr.io/
+ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-testing/debezium-testing-testcontainers/pom.xml
+++ b/debezium-testing/debezium-testing-testcontainers/pom.xml
@@ -35,6 +35,12 @@
             <!-- we need override default (i.e., test scope) to provided
             because we are using it within code in DebeziumContainer -->
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-slf4j-impl</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
@@ -92,6 +98,10 @@
             <artifactId>debezium-util</artifactId>
             <type>test-jar</type>
             <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-util</artifactId>
         </dependency>
 
         <dependency>

--- a/debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ImageNames.java
+++ b/debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ImageNames.java
@@ -13,13 +13,13 @@ public final class ImageNames {
 
     private static final String TIMESCALE_DB_IMAGE = "quay.io/debezium/timescale-timescaledb:latest-pg15";
 
+    public static final String OFFICIAL_MONGODB_IMAGE = "quay.io/debezium/official-mongo:8.0";
+
     public static final DockerImageName POSTGRES_DOCKER_IMAGE_NAME = DockerImageName.parse(POSTGRES_IMAGE)
             .asCompatibleSubstituteFor("postgres");
 
     public static final DockerImageName TIMESCALE_DB_IMAGE_NAME = DockerImageName.parse(TIMESCALE_DB_IMAGE)
             .asCompatibleSubstituteFor("postgres");
-
-    public static final String OFFICIAL_MONGODB_IMAGE = "mirror.gcr.io/library/mongo:6.0";
 
     public static final DockerImageName OFFICIAL_DOCKER_IMAGE_NAME = DockerImageName.parse(OFFICIAL_MONGODB_IMAGE)
             .asCompatibleSubstituteFor("mongodb");

--- a/debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/MongoDbContainer.java
+++ b/debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/MongoDbContainer.java
@@ -53,8 +53,8 @@ public class MongoDbContainer extends GenericContainer<MongoDbContainer> {
     /**
      * Default should match {@code version.mongo.server} in parent {@code pom.xml}.
      */
-    public static final String IMAGE_VERSION = System.getProperty("version.mongo.server", "6.0");
-    private static final DockerImageName IMAGE_NAME = DockerImageName.parse("mirror.gcr.io/library/mongo:" + IMAGE_VERSION);
+    public static final String IMAGE_VERSION = System.getProperty("version.mongo.server", "8.0");
+    private static final DockerImageName IMAGE_NAME = DockerImageName.parse("quay.io/debezium/official-mongo").withTag(IMAGE_VERSION);
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
     private static final String CONTAINER_KEYFILE_PATH = "/data/replica.key";

--- a/debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/MongoDbReplicaSet.java
+++ b/debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/MongoDbReplicaSet.java
@@ -21,8 +21,6 @@ import java.util.Optional;
 import java.util.Set;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
-import java.util.stream.StreamSupport;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -30,7 +28,6 @@ import org.testcontainers.containers.Container;
 import org.testcontainers.containers.Network;
 import org.testcontainers.images.ImagePullPolicy;
 import org.testcontainers.lifecycle.Startable;
-import org.testcontainers.shaded.com.fasterxml.jackson.databind.JsonNode;
 import org.testcontainers.utility.DockerImageName;
 import org.testcontainers.utility.MountableFile;
 
@@ -38,6 +35,7 @@ import io.debezium.testing.testcontainers.MongoDbContainer.Address;
 import io.debezium.testing.testcontainers.util.MoreStartables;
 import io.debezium.testing.testcontainers.util.PortResolver;
 import io.debezium.testing.testcontainers.util.RandomPortResolver;
+import io.debezium.util.Strings;
 
 /**
  * A MongoDB replica set.
@@ -372,23 +370,15 @@ public class MongoDbReplicaSet implements MongoDbDeployment {
     }
 
     public Optional<MongoDbContainer> tryPrimary() {
-        return stream(getStatus().path("members"))
-                .filter(memberStatus -> "PRIMARY".equals(memberStatus.path("stateStr").textValue()))
-                .findFirst()
-                .flatMap(this::findMember);
-    }
-
-    private Optional<MongoDbContainer> findMember(JsonNode memberStatus) {
-        var name = memberStatus.path("name").textValue();
+        var hello = members.get(0).eval("rs.hello()");
+        var primaryAddress = hello.path("primary").textValue();
+        if (Strings.isNullOrEmpty(primaryAddress)) {
+            return Optional.empty();
+        }
         return members.stream()
-                .filter(node -> node.getNamedAddress().toString().equals(name) || // Match by name or possibly IP
-                        node.getClientAddress().toString().equals(name))
+                .filter(node -> node.getNamedAddress().toString().equals(primaryAddress)
+                        || node.getClientAddress().toString().equals(primaryAddress))
                 .findFirst();
-    }
-
-    private JsonNode getStatus() {
-        var arbitraryNode = members.get(0);
-        return arbitraryNode.eval("rs.status()");
     }
 
     public List<String> getHostNames() {
@@ -408,10 +398,6 @@ public class MongoDbReplicaSet implements MongoDbDeployment {
                 ", members=" + members +
                 ", started=" + started +
                 '}';
-    }
-
-    private static <T> Stream<T> stream(Iterable<T> iterable) {
-        return StreamSupport.stream(iterable.spliterator(), false);
     }
 
 }

--- a/debezium-testing/debezium-testing-testcontainers/src/test/resources/testcontainers.properties
+++ b/debezium-testing/debezium-testing-testcontainers/src/test/resources/testcontainers.properties
@@ -1,1 +1,2 @@
+hub.image.name.prefix=quay.io/
 ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/debezium-testing/debezium-testing-testcontainers/src/test/resources/testcontainers.properties
+++ b/debezium-testing/debezium-testing-testcontainers/src/test/resources/testcontainers.properties
@@ -1,1 +1,1 @@
-hub.image.name.prefix=mirror.gcr.io/
+ryuk.container.image=quay.io/debezium/testcontainers-ryuk:0.13.0

--- a/pom.xml
+++ b/pom.xml
@@ -152,7 +152,7 @@
         <!-- Databases, should align with database drivers -->
         <version.mysql.server>8.2</version.mysql.server>
         <version.mysql.percona.server>8.0</version.mysql.percona.server>
-        <version.mongo.server>6.0</version.mongo.server>
+        <version.mongo.server>8.0</version.mongo.server>
         <version.cassandra3>3.11.12</version.cassandra3>
         <version.cassandra4>4.1.7</version.cassandra4>
         <version.cassandra5>5.0.2</version.cassandra5>

